# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [23]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [24]:
s1 = pd.Series(np.random.randint(1,5,size=100, dtype='l'))
s2 = pd.Series(np.random.randint(1, high=4, size=100, dtype='l'))
s3 = pd.Series(np.random.randint(10000, high=30001, size=100, dtype='l'))
print(s1,s2,s3)

0     4
1     3
2     1
3     1
4     1
     ..
95    1
96    4
97    1
98    3
99    2
Length: 100, dtype: int64 0     2
1     2
2     2
3     1
4     2
     ..
95    3
96    1
97    2
98    2
99    2
Length: 100, dtype: int64 0     24457
1     23793
2     13408
3     27424
4     26781
      ...  
95    16485
96    16703
97    14915
98    19532
99    18158
Length: 100, dtype: int64


### Step 3. Let's create a DataFrame by joinning the Series by column

In [25]:
df = pd.concat([s1,s2,s3],axis=1)

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [26]:
df.columns = ["bedrs","bathrs","price_sqr_meter"]
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   bedrs            100 non-null    int64
 1   bathrs           100 non-null    int64
 2   price_sqr_meter  100 non-null    int64
dtypes: int64(3)
memory usage: 2.5 KB


### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [27]:
big_column = pd.DataFrame(pd.concat([s1,s2,s3]),columns=["bigcolumn"])
big_column 

bigcolumn
0           4
1           3
2           1
3           1
4           1
..        ...
95      16485
96      16703
97      14915
98      19532
99      18158

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [28]:
big_column=big_column.reset_index(drop=True)
big_column

bigcolumn
0            4
1            3
2            1
3            1
4            1
..         ...
295      16485
296      16703
297      14915
298      19532
299      18158

[300 rows x 1 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299